client/network/src/block_request_handler.rs

```diff
@@ -121,12 +122,21 @@ impl <B: BlockT> BlockRequestHandler<B> {
 			let number = *header.number();
 			let hash = header.hash();
 			let parent_hash = *header.parent_hash();
-			let justification = if get_justification {
-				self.client.justification(&BlockId::Hash(hash))?
+			let justifications = if get_justification {
+				self.client.justifications(&BlockId::Hash(hash))?
 			} else {
 				None
 			};
-			let is_empty_justification = justification.as_ref().map(|j| j.is_empty()).unwrap_or(false);
+
+			// To keep compatibility we only send the grandpa justification
+			let justification = justifications
```

Comment:
Can't we just send the first one instead? Networking shouldn't depend on GRANDPA. Also we need a way to support serving the new justifications as well (at least eventually).

Reply:
Note on this: it was discussed elsewhere that we need to send and receive the GRANDPA ones, and not just the first one. To avoid the dependency on GRANDPA it was agreed to duplicate the ID tag here during a transition period

Reply:
> we need to send and receive the GRANDPA ones, and not just the first one

Why?

Reply:
Good point! I'll add a bit more here